### DISCUSSION TITLE
Should RDKit participate in Google Summer of Code?

### DISCUSSION THREAD

**Discussion by bertiewooster** [rdkit/rdkit#6009] [General] [answered]
I was wondering if there's interest in RDKit participating in [Google Summer of Code](https://summerofcode.withgoogle.com/) for 2023. The [deadline for organizations to apply is February 7](https://opensource.googleblog.com/2023/01/mentor-organization-applications-are-open-for-google-summer-of-code-2023.html). It's a program that pays a [stipend](https://google.github.io/gsocguides/student/why-should-i-apply.html) to [students or new developers](https://google.github.io/gsocguides/student/) to work with an open-source project, which provides mentors.

A maintainer of another open-source project wrote that organizations that have participated in the past (which [RDKit did as recently as 2022](https://summerofcode.withgoogle.com/archive/2020/projects/5646492108324864)) need only submit a project-ideas page and list how many mentors it has.

**Comment by greglandrum** [comment, marked as answer]
Hi @bertiewooster : We normally participate in GSoC as part of the OpenChemistry organization and I plan to do so again this year.
We will need project ideas and associated mentors, so if you have any ideas and want to volunteer, please let me know!

**Reply by bertiewooster** [reply]
Glad to know RDKit will participate. Is there an ideas page where people can contribute ideas? I noticed the [wiki](https://github.com/rdkit/rdkit/wiki) isn't editable by everyone, so maybe we should create a Google Doc? I'd be happy to if you like.

**Reply by greglandrum** [reply]
How about we just do it here, and then I will put the ideas into the openchemistry wiki once we've got them fleshed out?

**Comment by thegodone** [comment]
Hi both, I have a subject, extend PEOE partial charges parameters for more atoms or extend rdkit with another toolbox like https://github.com/krab1k/ChargeFW2Envoyé de mon iPhoneLe 27 janv. 2023 à 04:39, Greg Landrum ***@***.***> a écrit :﻿
How about we just do it here, and then I will put the ideas into the openchemistry wiki once we've got them fleshed out?

**Reply by bertiewooster** [reply]
I think your phone interfered with the link to that toolbox; it appears to be https://github.com/krab1k/ChargeFW2

**Reply by greglandrum** [reply]
Add a large and complex external depedency like ChargeFW2 isn't something I would support.

If there is a fast method for calculating charges which is as "good" as the current Gasteiger-Marsilli approach and which covers more of the periodic table, I'd be happy to see that implemented. I'm not deeply into this space, so pointers to references would be most welcome.

**Comment by thegodone** [comment]
I will take a look on lit.Envoyé de mon iPhoneLe 27 janv. 2023 à 06:15, Greg Landrum ***@***.***> a écrit :﻿
Add a large and complex external depedency like ChargeFW2 isn't something I would support.
If there is a fast method for calculating charges which is as "good" as the current Gasteiger-Marsilli approach and which covers more of the periodic table, I'd be happy to see that implemented. I'm not deeply into this space, so pointers to references would be most welcome.

**Comment by thegodone** [comment]
So my findings for 2D methods are list here :

Atomic Charge Calculator II: web-based tool for the calculation of partial
atomic charges
<https://www.researchgate.net/publication/341381167_Atomic_Charge_Calculator_II_web-based_tool_for_the_calculation_of_partial_atomic_charges>

[image: image.png]

MPEOE looks good for all P & S valences:
https://onlinelibrary.wiley.com/doi/abs/10.1002/jcc.540160808

This is used also a lot in diverse applications.

GG

**Comment by thegodone** [comment]
Indeed this thesis is very interesting on the correction adjustment of
paremeters of the MPEOE:

https://d-nb.info/978268148/34
